exercises/practice/sgf-parsing/.meta/src/SgfParsing.example.elm

```diff
@@ -0,0 +1,144 @@
+module SgfParsing exposing (Tree(..), parse)
+
+import Dict exposing (Dict)
+import Parser exposing ((|.), (|=), DeadEnd, Parser, Problem(..), Step(..), Trailing(..))
+
+
+type Tree
+    = Node { properties : Dict String (List String), children : List Tree }
+
+
+parse : String -> Result String Tree
+parse =
+    Parser.run (sgfWithParentheses |. Parser.end)
+        >> Result.mapError
+            (List.reverse >> List.head >> Maybe.map deadEndsToString >> Maybe.withDefault "")
+
+
+deadEndsToString : DeadEnd -> String
+deadEndsToString { problem } =
+    case problem of
+        ExpectingSymbol "(" ->
+            "tree missing"
+
+        ExpectingSymbol ";" ->
+            "tree with no nodes"
+
+        Expecting "[" ->
+            "properties without delimiter"
+
+        Problem string ->
+            string
+
+        _ ->
+            "Unexpected error"
+
+
+sgf : Parser Tree
+sgf =
+    Parser.oneOf [ sgfWithParentheses, sgfWithoutParentheses ]
+
+
+sgfWithParentheses : Parser Tree
+sgfWithParentheses =
+    Parser.succeed identity
```

Comment:
I haven't got my head fully in to this code, but I suspect that this can be written without `Parser.succeed identity`. Maybe just be removing it?

Reply:
I had the same feeling as you and I tried when I was implementing it, but `(|.) : Parser keep -> Parser ignore -> Parser keep` needs something to keep on its left, so it can't be `Parser.symbol "("`.
I tried to find something better without success. I just now found that the docs use the same approach [in one of the examples](https://package.elm-lang.org/packages/elm/parser/latest/Parser#keyword) so I guess it's probably fine.